src/index.js

```diff
@@ -1,104 +1,243 @@
-        return {id, name};
+        const { id, name } = element;
+        return { id, name };
       });
       return sortedFac;
     } catch (error) {
       console.log(error);
     }
   };
 
-  let choice;
+  const search = async (criteria) => {
+    try {
+      let charList = await getApi(
+        "characters?fields=*,equipments.equipments_id.*,factions.factions_id.*"
+      );
+      let found = charList.find((char) => char.name === criteria);
+      if (found) {
+        console.log(
+          `Oh, i can't believe it! Are you ${found.name} himself?\nPlease, come in!\n`
+        );
+        return found;
+      } else {
+        console.log("You are not registered in our army, sorry mate");
+      }
+    } catch (error) {
+      console.log(error);
+    }
+  };
 
-  while (choice != "0") {
-    choice = await read("1 - Create Character\n 2 - Log in\n 0 - Exit");
-    switch (choice) {
-      case "1":
-        console.log("Tell me traveller, who's side are you on?");
+  const create = async () => {
+    console.log("Tell me traveller, who's side are you on?");
+    let factionArray = await form();
+    console.log(`Welcome ${factionArray[0]} dev!`);
+
+    let newName = await read("What is your name?\n"); //example for read user input
+    clear();
+    console.log(`Nice to meet you ${newName}!\n`);
+    let persona = {
+      atk: 10,
+      def: 10,
+      agi: 10,
+      hp: 10,
+      gold: 100,
+      name: newName,
+      factions: [
+        {
+          factions_id: {
+            id: factionArray[1],
+            name: factionArray[0],
+          },
+        },
+      ],
+      equipments: [],
+    };
+    try {
+      const print = persona; //await postApi(persona, "characters?fields=*,equipments.equipments_id.*,factions.factions_id.*")
+    } catch (error) {
+      console.log(error);
+    }
+  };
+
+  const form = async () => {
+    let facChoice = 5;
+    try {
+      while (facChoice < 1 || facChoice > 3) {
         const factions = await getFactions();
-        factions.forEach((element, index) => {
-          console.log(index + " - " + element.name);
+        factions.forEach((facn, index) => {
+          console.log(`${index + 1} - ${facn.name}`);
         });
-
-        let faction;
-        let facId;
-        let facChoice = await read("");
-        switch (facChoice) {
-          case "0":
-            faction = factions[0].name;
-            facId = factions[0].id;
-            break;
-            case "1":
-              faction = factions[1].name;
-              facId = factions[1].id;
-              break;
-            case "2":
-              faction = factions[2].name;
-              facId = factions[2].id;
-            break;
-          default:
-            console.error("Chose a valid option");
-            break;
+        facChoice = await read("");
+        facChoice--;
+        if (facChoice >= 0 && facChoice < 3) {
+          return [factions[facChoice].name, factions[facChoice].id];
```

Comment:
suggestion, this function could return an object instead of an array. The values will be more meaningful when using the return of this function. `{id: factions[facChoice].id, name: factions[facChoice].name}`